mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -719,8 +724,15 @@ static LogicalResult printOperation(CppEmitter &emitter, emitc::CastOp castOp) {
   if (failed(emitter.emitAssignPrefix(op)))
     return failure();
   os << "(";
-  if (failed(emitter.emitType(op.getLoc(), op.getResult(0).getType())))
-    return failure();
+  // Cast of lvalues return lvalues and therefore references.
```

Comment:
This is an incursion of semantics into the translator, maybe introduce `resultIsReference`?